software/06.Nest/README.md

```diff
@@ -9,131 +9,200 @@
+ - What is a [controller](https://docs.nestjs.com/controllers)?
+ - What is a [module](https://docs.nestjs.com/modules)
+
+> ### 2.1 - poc-shop provider
+
+
+A provider, in the context of your project, refers to a class that offers specific services, functionality, or data manipulation related to the poc-shop feature and we will also see the dependency injections here.
+
+
+
+First of all you should import the `Injectable` decorator and then place on the class that you will also create named `PocShopService`.
+
+✔️ Validation:
+
+it should look like this:
+
+```ts
+@Injectable()
+export class PocShopService {}
 ```
 
-### Step 2.2: Use @Param() to retrieve URL route parameters
+> ### 2.2 - poc-shop controller
 
-First, read a little about [route parameters](https://docs.nestjs.com/controllers#route-parameters).
 
-You can now update the previously created `getArticle` function to take a title as a route `@Param()`.  
-Add debug logs again to check if you are correctly receiving it 🚀
+🎮 In this section, you're about to embark on an exciting journey of creating your first controller using the `@Controller('poc-shop')` decorator. 🚀 Controllers play a crucial role in handling incoming requests 📥 and responding to clients with the appropriate data. 💡
 
-> We'll use this param later to search for this title in our mocked article DB 😉
+To kick things off, let's dive into creating the PocShopController class. This class will serve as the controller specifically designed for the poc-shop route. 🛍️ Inside this class, we'll include a constructor that accepts an instance of the PocShopService and leverages the provider methods. 💼
 
-### Step 2.3: PUT request to update data
+But wait, there's more! To ensure the smooth flow of your application, you'll need to utilize the Get decorator.
 
-A PUT request usually uses both `@Param()` (e.g. find the article to update) and `@Body()` (containing the updated article)
+✔️ Validation:
 
-Using the `@Put()` method and what you have just learnt, write a route to get both the body and the url param.
+Now, let's take a look at the code snippet that brings it all together: 😃
 
-> As always, do not forget to test if everything works as intended before moving on to the next step!
+```ts
+@Controller('poc-shop')
+export class PocShopController {
+  constructor(private readonly pocShopService: PocShopService) {}
+}
+```
 
-### Step 2.4: Bonus
+> ### 2.3 - poc-shop module
 
-A controller usually contains routes with similar routes (e.g. `/user`, `/user/profile` ...).<br>
-With [controller routing](https://docs.nestjs.com/controllers#routing) you can specify a prefix to be applied to every routes of this controller (it would be `article` here)
+🔗 A module is a class annotated with a `@Module()` 🧩 **decorator**. The @Module() decorator provides metadata that Nest makes use of to organize the application `structure`.🏗️
 
-## Step 3: Database Service
+✔️ Validation:
```

Comment:
validation should not be a code correction, only text/screenshots describing what should be achieved